core/includes/form.inc

```diff
@@ -4994,7 +5039,7 @@ function theme_form_element_label($variables) {
   // If the element is required, a required marker is appended to the label.
   $required = !empty($element['#required']) ? theme('form_required_marker', array('element' => $element)) : '';
 
-  $title = filter_xss_admin($element['#title']);
+  $title = isset($element['#title']) ? filter_xss_admin($element['#title']) : '';
```

Comment:
I don't think that this is a good idea. Form elements should always specify a `'#title'`. If you don't want to display the title, then you can use `'#title_display' => 'invisible'`.

Reply:
...we seem to be doing something similar in `_form_validate()` by the way. There's 3 instances where we are setting the value of a string replacement to this:
```php
'!name' => empty($elements['#title']) ? $elements['#parents'][0] : $elements['#title']
```

Reply:
...seems that this calls for a helper function that does something along these lines:
```php
function _form_element_get_element_title($element) {
  return empty($element['#title']) ? $element['#parents'][0] : filter_xss_admin($element['#title']);
}
```

Reply:
...there's even more instances of this:
- 1 in `form_validate_color()`
- 1 in `form_validate_number()`

Reply:
@klonos if I didn't do this we got fatal errors in the tests. Something in the `switch ($element['#title_display']) {` section seems to have brought about this fatal error - my guess. Maybe the call to `$label = theme('form_element_label', $variables);` outside of the switch statement.

`#title` doesn't need to be set. See: https://docs.backdropcms.org/form_api where at least three elements don't have a `#title` attribute. Also `#type = 'item'`.

Reply:
I've reverted the `$label = theme('form_element_label', $variables);` change which was in the PR that I used as a starting point. I suspect that was causing the fatal error.

Reply:
That fixed it, and avoids the line I had added here as well.